Drafts of all User interface acceptance criteria. DO NOT MERGE

campbellssource opened the pull request:
This branch contains drafts for all acceptance criteria. We are prioritising the review an merge of a subset of these ACs, but this branch contains the all the latest merged ACs as well as drafts of the ones that may be merged later.

Easiest way to view them is to browser the branch here: https://github.com/vegaprotocol/specs/tree/Add-interface-acceptance-criteria/user-interface#user-interface-acceptance-criteria

SORD ONLY: https://github.com/vegaprotocol/specs/pull/1181
WITH only: https://github.com/vegaprotocol/specs/pull/1223

campbellssource committed:
roughing out the shape

campbellssource committed:
markdown formating fixes

campbellssource committed:
conent tidy up

campbellssource committed:
Mostly stubbing oout wallet page

campbellssource committed:
v1 of wallet ACs

campbellssource committed:
re-name wallet file

campbellssource committed:
tainting + other wallet improvements

campbellssource committed:
github hides lines that start with ...

campbellssource committed:
more formating fixes

campbellssource committed:
formatting

campbellssource committed:
more formating

campbellssource committed:
sub out staking

campbellssource committed:
staking update

campbellssource committed:
more staking

campbellssource committed:
associate back on it's own page

campbellssource committed:
associate on its own page

campbellssource committed:
more staking

campbellssource committed:
links to all started specs

campbellssource committed:
test AC code formating on github

campbellssource committed:
rename folder

campbellssource committed:
link to user-interface section from index readme

campbellssource committed:
re-phrase to include must in AC

campbellssource committed:
association done

campbellssource committed:
Staking draft V1

campbellssource committed:
governance tidy up

campbellssource committed:
for list of proposals

jtsang586 commented:
Overall, I like the level of detail throughout. The ACs are perfectly readable and pretty clear in how we expect features to behave. 
From a QA perspective I'm not entirely sure I find the `must` and `shoulds` that necessary now that I'm seeing them in a list. QA would write tests for every scenario listed if possible and naturally if a `must` step fails then a bunch of tests for the `shoulds` would also fail.

e.g. If associating tokens fails then I won't be able to see a balance of associated tokens ready to use

I wouldn't mind having the musts and shoulds merged to `should` as that's what we normally use to assert in Cypress.
And have `can` scenarios as features that we'd like to have but are not developed yet which will eventually become `should` when they are ready for testing?

campbellssource committed:
details re-uses list AC

campbellssource committed:
add network switcher placeholder

campbellssource committed:
fix broken code

campbellssource committed:
add stake as %

campbellssource committed:
Draft of make proposals- no codes

edd committed:
Rename 0000-NETW-network-and-nodes.md to 1006-NETW-network-and-nodes.md

0000 is not a valid spec code

edd committed:
Rename 0000-WALL-wallet.md to 1007-WALL-wallet.md

0000 is not a valid spec code

edd committed:
Rename 1007-WALL-wallet.md to 1012-WALL-wallet.md

Fix duplicate sequence number

edd committed:
Rename 1006-NETW-network-and-nodes.md to 1013-NETW-network-and-nodes.md

Fix duplicate sequence number

edd committed:
Renumber some codes

campbellssource committed:
rename wallet, fix readme

campbellssource committed:
fix wallet codes

campbellssource committed:
fix 2 digit code

campbellssource committed:
new asset proposals to have sub status

campbellssource committed:
some codes for vote

campbellssource committed:
codes on VOTE

campbellssource committed:
general proposal codes

campbellssource committed:
Codes on proposal type specifics

campbellssource committed:
removed some incomplete on Vote and Prop

campbellssource committed:
readme placeholder for transaction specs

campbellssource committed:
improved staking income

campbellssource committed:
codes for staking income

campbellssource committed:
vesting with codes

campbellssource committed:
Add files for wallet connect and txn

campbellssource committed:
mark wallet as incomplete

campbellssource committed:
Fix issues found by running approbation

campbellssource committed:
attempting to fix something odd

campbellssource committed:
typo

campbellssource committed:
fix link

campbellssource committed:
Fix some odd codes

campbellssource committed:
more fixes to codes

campbellssource committed:
final code error fixed

campbellssource committed:
remove duplicate pages

campbellssource committed:
draft vega wallet connect

campbellssource committed:
re-word input current vega address

campbellssource committed:
re-worded again to not force single key behaviour

campbellssource committed:
clarify about min amount of tokens req to propose

campbellssource committed:
added search to proposal listing

campbellssource committed:
added comment about LP requirement

campbellssource committed:
TODO: add more detail on change proposals

campbellssource committed:
improve readme

campbellssource committed:
wallet connect improvement

campbellssource committed:
add data format display appendix

campbellssource committed:
draft fills ACs

campbellssource committed:
add market data format

dexturr committed:
use html links instead of md

campbellssource committed:
Merge branch 'Add-interface-acceptance-criteria' of https://github.com/vegaprotocol/specs into Add-interface-acceptance-criteria

campbellssource committed:
test commit with signiture?

campbellssource committed:
this time?

campbellssource committed:
improved draft of SORD with WTXN, no codes

campbellssource committed:
WCON has codes

campbellssource committed:
fix code

campbellssource committed:
WTXN has codes

campbellssource committed:
WALL has codes

campbellssource committed:
spelling

campbellssource committed:
wrong codes

campbellssource committed:
fix some codes

campbellssource committed:
fix the codes

campbellssource committed:
content and code fixes

campbellssource committed:
titles on readme

campbellssource committed:
Submit order and prerequisites

campbellssource committed:
typo

campbellssource committed:
accept suggested improvement to wallet intro

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
accept suggested. prompt for update on *major* release

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
too 2 to

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
remove key from "key total balance"

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
accept suggestion: remove extra key

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
accept suggestion Vega -> a Vega wallet

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
accept spelling fix

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
accept suggestion: clarify un-taint steps

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
accept suggestion: clarify goal of getting a Vega wallet

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
Update user-interface/0001-WALL-wallet.md

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
Update user-interface/0001-WALL-wallet.md

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
Update user-interface/0001-WALL-wallet.md

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
Update user-interface/6001-SORD-submit_orders.md

Co-authored-by: m.ray <16125548+MadalinaRaicu@users.noreply.github.com>

campbellssource committed:
Update user-interface/0002-WCON-connect_vega_wallet.md

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
Update user-interface/0002-WCON-connect_vega_wallet.md

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
Update user-interface/0002-WCON-connect_vega_wallet.md

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
Update user-interface/0002-WCON-connect_vega_wallet.md

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
Update user-interface/0003-WTXN-submit_vega_transaction.md

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
Update user-interface/0003-WTXN-submit_vega_transaction.md

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
remove prompt when validators change

campbellssource committed:
improve context for updating wallet to compatible version

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
improve entering password once per "session"

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
clarify that you can see ignored transactions in the same places as pending, approved etc

campbellssource committed:
change searching for assets by market to a would

campbellssource committed:
0001-WALL-064 to only work for valid transactions

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
fix error code

campbellssource committed:
0001-WALL-030 search by matching markets and codes

campbellssource committed:
removed AC for getting recovery phrase add hoc

campbellssource committed:
Update user-interface/7001-DATA-data_display.md

campbellssource committed:
split revoke permisions into wallet and keys

campbellssource committed:
Merge branch 'user-interface-SORD-only' of https://github.com/vegaprotocol/specs into user-interface-SORD-only

campbellssource committed:
dapp that made a request is referer that can be spoofed

campbellssource committed:
added 0001-WALL-073 can't select tainted keys

campbellssource committed:
Merge branch 'user-interface-SORD-only' of https://github.com/vegaprotocol/specs into user-interface-SORD-only

campbellssource committed:
auto approve for x time or transactions

campbellssource committed:
Update user-interface/0003-WTXN-submit_vega_transaction.md

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
detail on how many decimal places to show

campbellssource committed:
add network to transaction history

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
add code for newest AC

campbellssource committed:
latest wallet download works for current mainnet

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
add code for latest AC

campbellssource committed:
save previous values for selected markets

campbellssource committed:
add move price by multiples by holding modifier key

campbellssource committed:
AC codes for moving values by larger steps

campbellssource committed:
add ability to enter values with "k" or "m" or "e2

campbellssource committed:
warn if validation not applicable for trading mode

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
add ac code for latest code

campbellssource committed:
add link to canceled order transaction in BE

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
Update user-interface/0002-WCON-connect_vega_wallet.md

campbellssource committed:
see trades on filled orders

Co-authored-by: Barnaby M <b.mannerings@gmail.com>

campbellssource committed:
short truncations of pub keys

campbellssource committed:
Update user-interface/7001-DATA-data_display.md

campbellssource committed:
Update user-interface/0001-WALL-wallet.md

campbellssource committed:
public keys don't start with 0x

campbellssource committed:
add transaction hash to data types

campbellssource committed:
reference transaction hash data type

campbellssource committed:
can't associate more than you have

campbellssource committed:
remove reference to ticket size

campbellssource committed:
typos

campbellssource committed:
Merge branch 'user-interface-SORD-only' of https://github.com/vegaprotocol/specs into user-interface-SORD-only

campbellssource committed:
more typo

campbellssource committed:
add some more estimates for order

campbellssource committed:
add indication of filled/remaining after order

campbellssource committed:
detail of nominations for associated tokens

campbellssource committed:
asset balances on key to a should

barnabee committed:
Apply suggestions from code review

Co-authored-by: Michael <michael@vega.xyz>

campbellssource committed:
the UIs not us

campbellssource committed:
Update user-interface/7001-DATA-data_display.md

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
Update user-interface/0002-WCON-connect_vega_wallet.md

campbellssource committed:
fairground wallet only on fairground

campbellssource committed:
Merge branch 'user-interface-SORD-only' of https://github.com/vegaprotocol/specs into user-interface-SORD-only

campbellssource committed:
0001-WALL-077 history from previous wallet

campbellssource committed:
0001-WALL-078 stop leaking keys info for wallet

campbellssource committed:
WALL-078 dont tell datanode what keys are inwallet

campbellssource committed:
make less solution specific

campbellssource committed:
wrap codes in brackets

ckpringle committed:
User interface AC tweaks (#1216)

* Update 0001-WALL-wallet.md

Small tweak to make the wording of downloading the wallet from .xyz less solution specific (we might need to change the solution there)

* Update 0002-WCON-connect_vega_wallet.md

Added - **should** have a way to easily access help should I need to troubleshoot my connection e.g. link to the docs (needs a link attached to it)

* Update 6001-SORD-submit_orders.md

A lot of this feels quite specific to the solution, but it also might not be worth changing this now if this is expected behaviour based on how users expect an exchange to work? 

For example:
input field pre-populated -> ... have the previous value for the selected market available e.g. pre-populated
hit up/down on the keyboard to increase -> ... be able to quickly tweak the size by the market's min-contract size e.g. hit up/down on the keyboard to increase

campbellssource committed:
+code for line about link to docs on get wallet

campbellssource committed:
Merge branch 'master' of https://github.com/vegaprotocol/specs into Add-interface-acceptance-criteria

campbellssource committed:
Merge branch 'master' of https://github.com/vegaprotocol/specs into user-interface-SORD-only

campbellssource committed:
Create Withdrawal page

campbellssource committed:
Add asset balances and account types

campbellssource committed:
add explanation to the wallets section works

campbellssource committed:
spelling

campbellssource committed:
Update user-interface/7001-DATA-data_display.md

campbellssource committed:
updated detail of scenario

campbellssource committed:
Merge branch 'user-interface-SORD-only' of https://github.com/vegaprotocol/specs into user-interface-SORD-only

campbellssource committed:
Update user-interface/6001-SORD-submit_orders.md

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
add detail about what a wallet does

campbellssource committed:
Merge branch 'user-interface-SORD-only' of https://github.com/vegaprotocol/specs into user-interface-SORD-only

campbellssource committed:
further correction to above

campbellssource committed:
typo

campbellssource committed:
typo

campbellssource committed:
added conditions appropriate for txn status

campbellssource committed:
actions to ignored transactions

campbellssource committed:
typo

PeteWilliams committed:
Apply suggestions from code review

Co-authored-by: Michael <michael@vega.xyz>

campbellssource committed:
improved draft up to completing transaction

campbellssource committed:
formatting

campbellssource committed:
connect to eth wallet

campbellssource committed:
link to connect eth wallet from readme

campbellssource committed:
fixes to wallet connect

campbellssource committed:
change wallet connect title and link to it

campbellssource committed:
Merge branch 'user-interface-SORD-only' of https://github.com/vegaprotocol/specs into user-interface-SORD-only

campbellssource committed:
draft submit eth transaction

campbellssource committed:
typos

campbellssource committed:
Add gas and approve to ETH txn

campbellssource committed:
corrections/ improvements

campbellssource committed:
post proof improvements

campbellssource committed:
spelling

campbellssource committed:
other oses too

campbellssource committed:
fixa code

campbellssource committed:
more code fixing

campbellssource committed:
moved set ticket inputs by leverage

campbellssource committed:
reading improvement

campbellssource committed:
Update user-interface/0001-WALL-wallet.md

Co-authored-by: candida-d <62548908+candida-d@users.noreply.github.com>

campbellssource committed:
add eager connection to wallet

campbellssource committed:
typo

campbellssource committed:
typo

campbellssource committed:
persistent transaction history

campbellssource committed:
latest SORD with Everything branch

campbellssource committed:
latest WITH into everything branch

campbellssource committed:
remove withdraw incomplete label

campbellssource committed:
replace missing tittle

campbellssource committed:
stub for TRANsfers

campbellssource committed:
stub Market DETails

campbellssource committed:
some manage order notes

campbellssource committed:
more draft on MORD

campbellssource committed:
add side to DATA

campbellssource committed:
mord rough draft

campbellssource commented:
Files have been significantly changed and the root that lists all the files is now on master so don't need to keep this open anymore
